### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by shodgesio**
Follow-up to #3 

In order to best expose the API and the expected configuration to be passed into the RefTagger, I've converted the source to TypeScript and created interfaces representing what documentation I am able to find at https://faithlife.com/products/reftagger/customize.

Additional configuration was added so that NPM builds and provides this package as JavaScript with typings included. The resulting package can be consumed by projects written in either JavaScript or TypeScript, and the interfaces will be exposed for consumers to see.

Additionally, having some typing seemed to align better with Faithlife's heavy use of the .NET environment internally, which I also work in regularly.

### CODE REVIEW

**Comment by shodgesio** [comment]
@davidschwegler , here's a follow-up to our conversation on #3 . Thanks for your implementation of those changes!

File: package.json [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "files": [
+    "dist/**/*"
+  ],
   "scripts": {
-    "test": "test"
+    "build": "tsc",
+    "format": "prettier --write \"src/**/*.ts\"",
+    "lint": "tslint -p tsconfig.json",
```
Comment (ddunkin) [inline package.json:13]: We should use our standard eslint configuration. https://www.npmjs.com/package/eslint-config-faithlife

Comment (shodgesio) [reply package.json:13]: @ddunkin , I'll incorporate that. Thanks!

**Comment by davidschwegler** [comment]
Thanks for this! 

Wanted to give you an update on the timeline -- my team is in the middle of a big push right now, so I won't be able to test and merge this for a few weeks (unless someone beats me to it).

**Comment by shodgesio** [comment]
> Thanks for this!
> 
> Wanted to give you an update on the timeline -- my team is in the middle of a big push right now, so I won't be able to test and merge this for a few weeks (unless someone beats me to it).

No worries! This was more of a fun Wednesday afternoon follow-up to our last conversation, and less of an urgent need. All the best to you guys.

**Comment by davidschwegler** [comment]
Hey @shodgesio, I didn't forget about you! By "less of an urgent need", I assume you _definitely_ meant "preferably sometime before Jesus returns", right? ;) I wish I had gotten to it sooner, though.

Let me know if there's any issues with this release.

**Comment by shodgesio** [comment]
Thanks so much @davidschwegler !

**Pull request merged**